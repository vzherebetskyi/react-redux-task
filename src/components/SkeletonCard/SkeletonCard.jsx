import React from 'react';
import Skeleton from 'react-loading-skeleton'
import 'react-loading-skeleton/dist/skeleton.css'

import './SkeletonCard.scss';

const SkeletonCard = () => (
    <div className='card-wrapper'>
      <p className='card-title'><Skeleton /></p>
        <Skeleton height={100} />
      <div className='card-info-container'>
        <div className='card-info-container__author'>
          <p><Skeleton /></p>
        </div>
        <div className='card-info-container__published'>
          <p><Skeleton /></p>
        </div>
        <div className='card-info-container__source'>
          <p><Skeleton /></p>
        </div>
      </div>
      <div className='card-description-container'>
        <p><Skeleton count={3} /></p>
        <p><Skeleton /></p>
      </div>
    </div>
  );

export default SkeletonCard;
