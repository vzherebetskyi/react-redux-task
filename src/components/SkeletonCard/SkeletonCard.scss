.card-wrapper {
  height: 20rem;
  width: 100%;
  max-width: 40rem;
  margin: 10px auto;
  border: 1px solid lightgray;
  border-radius: 5px;
  padding: 10px;

  @media screen and (max-width: 1000px) {
    height: 18rem;
    width: 90vw;
  }
}

.card-title {
  margin: 0;
  font-weight: bold;
  font-size: 1.2rem;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;

  @media screen and (max-width: 1000px) {
    font-size: 1rem;
  }
}

.card-image-container {
  width: 100%;
  display: flex;
  justify-content: center;
}

.card-image {
  height: 10rem;
  width: 90%;
  object-fit: cover;

  @media screen and (max-width: 1000px) {
    height: 8rem;
  }
}

.card-info-container {
  display: flex;

  div {
    min-width: 0;
    margin-top: 0.5rem;
    
    span {
      font-weight: bold;
    }
  }

  p {
    margin: 0;
    font-size: 0.9rem;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;

    @media screen and (max-width: 1000px) {
      font-size: 0.7rem;
    }
  }

  &__author {
    flex: 1;
  }

  &__published {
    flex: 1;
  }

  &__source {
    flex: 1;
  }

  &__api {
    flex: 1;
  }
}

.card-description-container {
  @media screen and (max-width: 1000px) {
    font-size: 0.9rem;
  }
}
