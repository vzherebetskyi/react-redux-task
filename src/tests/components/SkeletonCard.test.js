import React from 'react';
import renderer from 'react-test-renderer';

import SkeletonCard from '../../components/SkeletonCard/SkeletonCard';

test('should render SkeletonCard component correctly', () => {
  const elem = renderer.create(<SkeletonCard />).toJSON();
  expect(elem).toMatchSnapshot();
});
