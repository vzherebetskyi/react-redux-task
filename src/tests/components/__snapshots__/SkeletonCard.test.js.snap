// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SkeletonCard component correctly 1`] = `
<div
  className="card-wrapper"
>
  <p
    className="card-title"
  >
    <span
      aria-busy={true}
      aria-live="polite"
    >
      <span
        className="react-loading-skeleton"
        style={{}}
      >
        ‌
      </span>
      <br />
    </span>
  </p>
  <span
    aria-busy={true}
    aria-live="polite"
  >
    <span
      className="react-loading-skeleton"
      style={
        {
          "height": 100,
        }
      }
    >
      ‌
    </span>
    <br />
  </span>
  <div
    className="card-info-container"
  >
    <div
      className="card-info-container__author"
    >
      <p>
        <span
          aria-busy={true}
          aria-live="polite"
        >
          <span
            className="react-loading-skeleton"
            style={{}}
          >
            ‌
          </span>
          <br />
        </span>
      </p>
    </div>
    <div
      className="card-info-container__published"
    >
      <p>
        <span
          aria-busy={true}
          aria-live="polite"
        >
          <span
            className="react-loading-skeleton"
            style={{}}
          >
            ‌
          </span>
          <br />
        </span>
      </p>
    </div>
    <div
      className="card-info-container__source"
    >
      <p>
        <span
          aria-busy={true}
          aria-live="polite"
        >
          <span
            className="react-loading-skeleton"
            style={{}}
          >
            ‌
          </span>
          <br />
        </span>
      </p>
    </div>
  </div>
  <div
    className="card-description-container"
  >
    <p>
      <span
        aria-busy={true}
        aria-live="polite"
      >
        <span
          className="react-loading-skeleton"
          style={{}}
        >
          ‌
        </span>
        <br />
        <span
          className="react-loading-skeleton"
          style={{}}
        >
          ‌
        </span>
        <br />
        <span
          className="react-loading-skeleton"
          style={{}}
        >
          ‌
        </span>
        <br />
      </span>
    </p>
    <p>
      <span
        aria-busy={true}
        aria-live="polite"
      >
        <span
          className="react-loading-skeleton"
          style={{}}
        >
          ‌
        </span>
        <br />
      </span>
    </p>
  </div>
</div>
`;
